CASMTRIAGE-3591 kernel dump fix (#2004)
Cray-HPE/docs-csm: scripts/hotfixes/kdump/README.md

# Kernel Dump Hotfix

> This hotfix comes from (`CASMTRIAGE-3591`)[https://jira-pro.its.hpecorp.net:8443/browse/CASMTRIAGE-3591].

Reviewer: Agreed. I don't think we should link to internal Jiras in the customer-facing docs, since they won't be able to follow the link. If we just want to provide a breadcrumb for internal folks, I think that makes more sense as a comment in the script.

> This hotfix comes from (`CASMTRIAGE-3591`)[https://jira-pro.its.hpecorp.net:8443/browse/CASMTRIAGE-3591].

Author: Moved to script.

    ```

   Example output:

   ```text
   Uploading hotfix files to ncn-m001:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-m002:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-m003:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-s001:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-s002:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-s003:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-s004:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-w001:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-w002:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-w003:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-w004:/srv/cray/scripts/common/ ... Done
   Running updated create-kdump-artifacts.sh script on [11] NCNs ... Done
   The following NCNs contain the kdump patch:
   ncn-m001
   ncn-m002
   ncn-m003
   ncn-s001
   ncn-s002
   ncn-s003
   ncn-s004
   ncn-w001
   ncn-w002
   ncn-w003
   ncn-w004
   This hotfix has completed.
   ```

## Background

> The original script and library used live in [metal-provision](https://github.com/Cray-HPE/metal-provision/blob/main/roles/ncn-common-setup/files/srv/cray/scripts/common/create-kdump-artifacts.sh)

Reviewer: ```suggestion
The original script and library came from [metal-provision](https://github.com/Cray-HPE/metal-provision/blob/main/roles/ncn-common-setup/files/srv/cray/scripts/common/create-kdump-artifacts.sh).
```

> The only difference between the script in this hotfix and the script for CSM 1.3 should be the inclusion/exclusion of

Reviewer: This seems like an odd thing to mention. Also doesn't really feel like "Background". I'd omit this part.

Author: Leaving it in.

Reviewer: How is it helpful for customers to have this here? If it's here for the benefit of developers, it seems like the wrong place for it.

Author: Why does this matter so much? This is mainly emphasis that this script and the perm fix is the same, and for record keeping it shouldn't differ or its differences should be very explicit.

Author: is the issue with the clause mentioning the 3 differences?

Author: removed the clause.

Reviewer: Yeah, I was talking here specifically about the 3 differences line. If just wanting to emphasize that this is essentially the same as the permanent fix, I'd say something closer to exactly that -- "This hot fix is essentially the same as the proper fix that is being introduced in CSM 1.3."

Author: Makes sense.

   Uploading hotfix files to ncn-m002:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-m003:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-s001:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-s002:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-s003:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-s004:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-w001:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-w002:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-w003:/srv/cray/scripts/common/ ... Done
   Uploading hotfix files to ncn-w004:/srv/cray/scripts/common/ ... Done
   Running updated create-kdump-artifacts.sh script on [11] NCNs ... Done
   The following NCNs contain the kdump patch:
   ncn-m001
   ncn-m002
   ncn-m003
   ncn-s001
   ncn-s002
   ncn-s003
   ncn-s004
   ncn-w001
   ncn-w002
   ncn-w003
   ncn-w004
   This hotfix has completed.
   ```

## Origin

The original script and library used live in [metal-provision](https://github.com/Cray-HPE/metal-provision/tree/v1.0.6):

> - [`create-kdump-artifacts.sh`](https://github.com/Cray-HPE/metal-provision/blob/v1.0.6/roles/ncn-common-setup/files/srv/cray/scripts/common/create-kdump-artifacts.sh)

Reviewer: ```suggestion

- [`create-kdump-artifacts.sh`](https://github.com/Cray-HPE/metal-provision/blob/v1.0.6/roles/ncn-common-setup/files/srv/cray/scripts/common/create-kdump-artifacts.sh)
```

Author: Fixed in parallel.
